src/util/SaveNameUtils.cpp

```diff
@@ -0,0 +1,45 @@
+#include "util/SaveNameUtils.h"
+
+#include <sstream>            // for stringstream
+#include <chrono>             // for time
+#include <iomanip>            // for put_time
+
+#include "util/PathUtil.h"    // for clearExtensions
+
+
+auto SaveNameUtils::parseFilenameFromWildcardString(const std::string& wildcardString, fs::path defaultFilePath) -> std::string {
+    std::string saveString = wildcardString;
+    size_t pos = saveString.find(DEFAULT_WILDCARD_START);
+
+    // parse all wildcards until none are left
+    while (pos != std::string::npos) {
+        size_t wildcardStartLength = std::strlen(DEFAULT_WILDCARD_START);
+        size_t endPos = saveString.find(DEFAULT_WILDCARD_END, pos + wildcardStartLength);
+        if (endPos == std::string::npos) {
+            break;
+        }
+        std::string wildcard = saveString.substr(pos + wildcardStartLength, endPos - pos - wildcardStartLength);
+        saveString.replace(pos, endPos + 1 - pos, parseWildcard(wildcard, defaultFilePath));
+        pos = saveString.find(DEFAULT_WILDCARD_START, pos);
+    }
+
+    return saveString;
+}
+
+auto SaveNameUtils::parseWildcard(const std::string& wildcard, fs::path defaultFilePath) -> std::string {
+    if (wildcard == WILDCARD_NAME) {
+        Util::clearExtensions(defaultFilePath, ".pdf");
+        return defaultFilePath.u8string();
+    }
```

Comment:
Only copy the path when actually needed.

```suggestion
auto SaveNameUtils::parseWildcard(const std::string& wildcard, const fs::path& defaultFilePath) -> std::string {
    if (wildcard == WILDCARD_NAME) {
        fs::path path = defaultFilePath;
        Util::clearExtensions(path, ".pdf");
        return path.u8string();
    }
```